### PR TITLE
Fix the percentile argument for ApproxPercentileCont must be Float64, not Decimal128(2, 1)

### LINKED ISSUE #4103: The percentile argument for ApproxPercentileCont must be Float64, not Decimal128(2, 1)
**Describe the bug**
I found this when testing https://github.com/apache/arrow-datafusion/pull/4102. `csv_query_approx_percentile_cont` fails with:

```
The percentile argument for ApproxPercentileCont must be Float64, not Decimal128(2, 1)
```

**To Reproduce**
Run tests in https://github.com/apache/arrow-datafusion/pull/4102

**Expected behavior**
I assume we need to add type coercion to convert function args to the correct type

**Additional context**

### PR DESCRIPTION
**Pull request by comphead**
# Which issue does this PR close?

Closes #4103 .

# Rationale for this change
Fix the percentile argument for ApproxPercentileCont must be Float64, not Decimal128(2, 1)

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by comphead** [comment]
@andygrove please check the PR

**Comment by andygrove** [comment]
LGTM @comphead, but I think this PR also needs a unit test so that we don't have regressions in the future.

**Comment by comphead** [comment]
Added test

File: datafusion/physical-expr/src/aggregate/tdigest.rs [hunk 1]
```diff
@@ -231,7 +231,6 @@ impl TDigest {
     }
 
     pub(crate) fn merge_sorted_f64(&self, sorted_values: &[f64]) -> TDigest {
-        dbg!(&sorted_values);
```
Comment (alamb) [inline datafusion/physical-expr/src/aggregate/tdigest.rs:234]: 👍

File: datafusion/expr/src/type_coercion/aggregates.rs [hunk 2]
```diff
@@ -166,19 +168,22 @@ pub fn coerce_types(
...
                     )));
             }
-            Ok(input_types.to_vec())
+            let mut result = input_types.to_vec();
+            if can_coerce_from(&DataType::Float64, &input_types[1]) {
+                result[1] = DataType::Float64;
+            } else {
+                return Err(DataFusionError::Plan(format!(
+                    "The percentile argument for {:?} must be Float64, not {:?}.",
+                    agg_fun, input_types[1]
```
Comment (alamb) [inline datafusion/expr/src/type_coercion/aggregates.rs:183]:
```suggestion
                    "Could not coerce the percent argument for {:?} to Float64. Was  {:?}.",
                    agg_fun, input_types[1]
```

**Review by alamb — APPROVED** [review]
Looks good to me -- thanks @comphead

File: datafusion/expr/src/type_coercion/aggregates.rs (hunk 2 above)
Comment (comphead) [reply datafusion/expr/src/type_coercion/aggregates.rs]: Done

**Comment by alamb** [comment]
Thanks @comphead

**Pull request merged**